CONTRIBUTING.rst

```diff
@@ -0,0 +1,678 @@
+
+.. code-block:: bash
+
+    pre-commit run
+
+
+-   Run only mypy check on your staged files by using:
+
+.. code-block:: bash
+
+    pre-commit run mypy
+
+
+-   Run only mypy checks on all files by using:
+
+.. code-block:: bash
+
+    pre-commit run mypy --all-files
+
+
+-   Run all checks on all files by using:
+
+.. code-block:: bash
+
+    pre-commit run --all-files
+
+
+-   Skip one or more of the checks by specifying a comma-separated list of
+    checks to skip in the SKIP variable:
+
+.. code-block:: bash
+
+    SKIP=pylint,mypy pre-commit run --all-files
+
+
+You can always skip running the tests by providing ``--no-verify`` flag to the
+``git commit`` command.
+
+To check other usage types of the pre-commit framework, see `Pre-commit website <https://pre-commit.com/>`__.
+
+Travis CI Testing Framework
+===========================
+
+Airflow test suite is based on Travis CI framework as running all of the tests
+locally requires significant setup. You can set up Travis CI in your fork of
+Airflow by following the `Travis
+CI Getting Started
+guide <https://docs.travis-ci.com/user/getting-started/>`__..
+
+
+There are two different options available for running Travis CI, and they are
+set up on GitHub as separate components:
+
+-   **Travis CI GitHub App** (new version)
+-   **Travis CI GitHub Services** (legacy version)
+
+Using Travis CI GitHub App (new version)
+----------------------------------------
+
+-   Once `installed <https://github.com/apps/travis-ci/installations/new/permissions?target_id=47426163>`__,
+    configure the Travis CI GitHub App at
+    `Configure Travis CI <https://github.com/settings/installations>`__.
+
+-   Set repository access to either "All repositories" for convenience, or "Only
+    select repositories" and choose ``USERNAME/airflow`` in the drop-down menu.
+
+-   Access Travis CI for your fork at `<https://travis-ci.com/USERNAME/airflow>`__.
+
+Using Travis CI GitHub Services (legacy version)
+------------------------------------------------
+
+..
+
+    The apache/airflow project is still using the legacy version.
+
+Travis CI GitHub Services version uses an Authorized OAuth App.
+
+1.  Once installed, configure the Travis CI Authorized OAuth App at
+    `Travis CI OAuth APP <https://github.com/settings/connections/applications/88c5b97de2dbfc50f3ac>`__.
+
+2.  If you are a GitHub admin, click the **Grant** button next to your
+    organization; otherwise, click the **Request** button. For the Travis CI
+    Authorized OAuth App, you may have to grant access to the forked
+    ``ORGANIZATION/airflow`` repo even though it is public.
+
+3.  Access Travis CI for your fork at
+    `<https://travis-ci.org/ORGANIZATION/airflow>`_.
+
+Creating New Projects in Travis CI
+----------------------------------
+
+If you need to create a new project in Travis CI, use travis-ci.com for both
+private repos and open source.
+
+The travis-ci.org site for open source projects is now legacy and you should not use it.
```

Comment:
Elena: why may the contributors need to create a projecy in Travis CI?
Jarek: They do not need it. I updated the description.